File: src/me/krickl/memebotj/InternalCommands/AdminCommands/QuitCommand.scala
package me.krickl.memebotj.InternalCommands.AdminCommands

import me.krickl.memebotj.Utility.CommandPower
import me.krickl.memebotj._

class QuitCommand(channel: String, command: String, dbprefix: String) extends CommandHandler(channel,
  command, dbprefix) {

  this.setNeededCommandPower(CommandPower.adminAbsolute)

  this.setUnformattedOutput(Memebot.formatText("QUIT", channelOriginHandler, null, this, true, Array()))

  this.setHelptext(Memebot.formatText("QUIT_SYNTAX", channelOriginHandler, null, this, true, Array()))

  override def commandScript(sender: UserHandler, channelHandler: ChannelHandler, data: Array[String]) {
    System.exit(0)
  }
}

File: src/me/krickl/memebotj/Commands/Internal/QuitCommand.java
package me.krickl.memebotj.Commands.Internal;

import me.krickl.memebotj.ChannelHandler;
import me.krickl.memebotj.Commands.CommandHandler;
import me.krickl.memebotj.Memebot;
import me.krickl.memebotj.UserHandler;
import me.krickl.memebotj.Utility.CommandPower;

/**
 * This file is part of memebotj.
 * Created by unlink on 11/04/16.
 */
public class QuitCommand extends CommandHandler {
    public QuitCommand(ChannelHandler channelHandler, String commandName, String dbprefix) {
        super(channelHandler, commandName, dbprefix);
    }

    @Override
    public void overrideDB() {
        this.setNeededCommandPower(CommandPower.adminAbsolute);

        this.setUnformattedOutput(Memebot.formatText("QUIT", getChannelHandler(), null, this, true, new String[]{}, ""));

        this.setHelptext(Memebot.formatText("QUIT_SYNTAX", getChannelHandler(), null, this, true, new String[]{}, ""));
    }

    @Override
    public void commandScript(UserHandler sender, String[] data) {
        System.exit(0);
    }
}
